### PR TITLE
Add placeholder MMAction and MMReduxAction types

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
Continuing on from #29483, this includes more type changes to add new `MMAction` and `MMReduxAction` types to the general web app and mattermost-redux. Similar to the last PR, this one is almost exclusively type changes. There's a few changes to test code which aren't 100% needed now, but they'll be needed when we start to define those action types next.

The reason those types are still just placeholders is because I wanted to get all the name changes out of the way before the actual changes that matter to make it so that we can more strongly type Redux actions.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/packages/mattermost-redux/src/reducers/entities/cloud.test.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ const minimalLimits = {
 
 describe('limits reducer', () => {
     test('returns empty limits by default', () => {
-        expect(limits(undefined, {type: 'some action', data: minimalLimits})).toEqual({
+        expect(limits(undefined, {type: undefined})).toEqual({
```
Comment (larkox) [inline webapp/channels/src/packages/mattermost-redux/src/reducers/entities/cloud.test.ts:19]: Why these changes?

File: webapp/channels/src/packages/mattermost-redux/src/reducers/websocket.ts [hunk 2]
```diff
@@ -33,10 +32,6 @@ export default function reducer(state = getInitialState(), action: AnyAction) {
         return getInitialState();
     }
 
-    if (action.type === UserTypes.LOGOUT_SUCCESS) {
```
Comment (larkox) [inline webapp/channels/src/packages/mattermost-redux/src/reducers/websocket.ts:36]: Why is this removed?

**Review by larkox — APPROVED** [review]
LGTM.

There are a couple of changes in tests that I don't completely get (probably just fixing types) and one thing that got removed that seems off, but apart of that, looks great.

File: webapp/channels/src/packages/mattermost-redux/src/reducers/entities/cloud.test.ts (hunk 1 above)
Comment (hmhealey) [reply webapp/channels/src/packages/mattermost-redux/src/reducers/entities/cloud.test.ts:19]: I didn't split this part out quite right when I broke up these changes since it isn't technically needed until #29486.

My end goal with this set of changes is to make it so that we can define types for some actions and treat the rest like `AnyAction` as they do currently, but unfortunately to do that, I had to list every possible value of `type` which means that we can't dispatch arbitrary actions here to initialize the store. I had also tried to make this an empty string since that feels more correct, but it caused problems with type inference, so I went with `{type: undefined}` which some of these tests already used

File: webapp/channels/src/packages/mattermost-redux/src/reducers/websocket.ts (hunk 2 above)
Comment (hmhealey) [reply webapp/channels/src/packages/mattermost-redux/src/reducers/websocket.ts:36]: There's actually an exact copy of this directly above it. I think it must've been introduced by a bad merge at some point, but we never noticed it. I only came across it because TS starts complaining once the changes in #29486 are made since it knows that `action.type` can't have that value at this point

**Review by pvev — APPROVED** [review]
LGTM! 🚀
